crates/runtime/src/memory.rs

```diff
@@ -246,7 +251,8 @@ impl Memory {
     pub fn new_static(
         plan: &MemoryPlan,
         base: &'static mut [u8],
-        make_accessible: fn(*mut u8, usize) -> Result<()>,
+        make_accessible: fn(PoolingBackend, *mut u8, usize) -> Result<()>,
```

Comment:
And another possible way to organize this is something like:

```rust
enum PoolingBackend {
    Mmap(NativeMmap),
    Uffd(Uffd),
    Memfd(Memfd),
}

struct NativeMmap;

impl NativeMmap {
    // methods here, probably one impl of `NativeMmap` per-OS
}

enum Memfd {
    #[cfg(feature = "memfd")]
    Supported,
}

impl Memfd {
    fn new() -> Result<Memfd> {
        #[cfg(feature = "memfd")]
        return Ok(Memfd::Supported);
        #[cfg(not(feature = "memfd"))]
        bail!("support not compiled in (or os not supported");
    }

    fn some_method(&self) -> Result<()> {
        match self {
            #[cfg(feature = "memfd")]
            Memfd::Supported => {
                // the implementation
            }
        }
    }
}
```

I think this would be a more manageable way to handle platform-specific and feature-specific functionality perhaps?